nemo/collections/nlp/modules/common/transformer/text_generable_model.py

```diff
@@ -0,0 +1,59 @@
+from typing import List, TypedDict, Union
+
+from torch import Tensor
+
+
+class SamplingParam(TypedDict):
+    max_length: int  # The maximum length of the sequence to be generated.
+    min_length: int  # The minimum length of the sequence to be generated.
+    do_sample: bool  # Whether or not to use sampling ; use greedy decoding otherwise
+    top_k: int  # The number of highest probability vocabulary tokens to keep for top-k-filtering.
+    top_p: float  # If set to float < 1, only the most probable tokens with probabilities that add up to top_p or higher are kept for generation.
+    repetition_penalty: float  # The parameter for repetition penalty. 1.0 means no penalty.
+    add_BOS: bool  # add the bos token at the begining of the prompt
+
+
+class OutputType(TypedDict):
+    sentences: List[str]  # output sentences
+    tokens: List[List[str]]  # output sentences borken into tokens
+    logits: List[Tensor]  # logits output for all the tokens
+    token_ids: List[Tensor]  # output sentence token ids
+
+
+class TextGenerableModelInterface:
```

Comment:
Class naming conventions were described in our internal dev guide but I moved it into https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md so that anyone can see it. 

Let's rename this class to something like ``TextGeneration``

Reply:
changed